pyomo/core/base/reference.py

```diff
@@ -518,13 +521,19 @@ def Reference(reference, ctype=_NotSpecified):
         pass
     elif isinstance(reference, Component):
         reference = reference[...]
+    elif isinstance(reference, ComponentData):
+        obj = reference.type()()
+        obj.construct()
+        obj._data = {None: reference}
+        return obj
```

Comment:
I reverted to returning a SimpleComponent containing the ```ComponentData``` because the alternative broke my integrator somewhere in the nl file write. This is closer to how References are currently handled for SimpleComponents and for some reason doesn't break my integrator.